applications/FluidDynamicsApplication/tests/Cavity/cfd_function_process.json

```diff
@@ -0,0 +1,217 @@
+                        "-2*t*x*x",
+                        "t*y*3*y"
+                    ],
+                    "interval": [
+                        0.0,
+                        "End"
+                    ]
+                }
+            },
+            {
+                "python_module": "assign_scalar_variable_process",
+                "kratos_module": "KratosMultiphysics",
+                "process_name": "AssignScalarVariableProcess",
+                "Parameters": {
+                    "model_part_name": "MainModelPart.Parts_Fluid",
+                    "variable_name": "VISCOSITY",
+                    "value": "t*t+2",
+                    "constrained": false
+                }
+            },
+            {
+                "python_module": "assign_scalar_variable_process",
+                "kratos_module": "KratosMultiphysics",
+                "process_name": "AssignScalarVariableProcess",
+                "Parameters": {
+                    "model_part_name": "MainModelPart.Parts_Fluid",
+                    "variable_name": "DENSITY",
+                    "value": "t*t*3+2",
+                    "constrained": false
+                }
+            }
+        ],
+        "gravity": [],
+        "auxiliar_process_list": [
+            {
+                "kratos_module": "KratosMultiphysics.FluidDynamicsApplication",
+                "python_module": "cfd_function_process",
+                "help": "",
+                "Parameters": {
+                    "model_part_name": "MainModelPart.NoSlip2D_left_wall",
+                    "function_settings": {
+                        "function_name": "CalculateYPlusAndUTauForConditionsBasedOnReaction",
+                        "kinematic_viscosity_variable_name": "VISCOSITY",
+                        "reaction_variable_name": "REACTION"
+                    }
+                }
+            },
+            {
+                "kratos_module": "KratosMultiphysics.FluidDynamicsApplication",
+                "python_module": "cfd_function_process",
+                "help": "",
+                "Parameters": {
+                    "model_part_name": "MainModelPart.NoSlip2D_right_wall",
+                    "function_settings": {
+                        "function_name": "CalculateYPlusAndUTauForConditionsBasedOnLinearLogarithmicWallFunction",
+                        "kinematic_viscosity_variable_name": "VISCOSITY",
+                        "von_karman": 0.41,
+                        "wall_smoothness": 5.2,
+                        "max_iterations": 20,
+                        "tolerance": 1e-6
+                    }
+                }
+            },
+            {
+                "kratos_module": "KratosMultiphysics.FluidDynamicsApplication",
+                "python_module": "cfd_function_process",
+                "help": "",
+                "Parameters": {
+                    "model_part_name": "MainModelPart",
+                    "function_settings": {
```

Comment:
same, theses should be separated processes...i see no reason for having a nested function_settings

Reply:
The reason behind this is, if I am to use different processes, then I need to have different python files which will have same code redundancy. I wanted to reduce code redundancy as much as possible.

Reply:
the point is that doing the way you do it is non-standard.

wouldn't it be possible to have a base_cfd_utility_process which implements the common code, and to derive all other processes from that one?

Reply:
Thats possible. I will change it like that :)

Reply:
THB I don't love the idea of implementing an additional concept in parallel to the processes. I.e. this could easily be done with separate processes (for which several py-scripts are needed ofc).

Is the only reason to do it as you propose bcs you don't want the additional py-scripts? Or would you have additional overhead?

To me the current script (cfd_function_process.py) is completely unreadable :/

If we really want to go this way then why not what is done in the HDF or the statistics app? There it is cleaner IMO

Reply:
I understand your point, as you stated, only reason I did it like this is to avoid maintaining several py files, make it easier for extending for new methods easily, keep list of files in check inside `python_scripts folder`.  I don't think there is an additional overhead if I do seperate py files, except for overhead in maintaining them.

Could I use folder sctructure in python_scripts with __init__.py? @philbucher @RiccardoRossi @rubenzorrilla

Reply:
> Could I use folder sctructure in python_scripts with __init__.py

yes, see the CoSimApp

Reply:
we can speak in person tmr, I think this way we can eliminate some concerns